pkg/cri/resource-manager/policy/builtin/topology-aware/resources.go

```diff
@@ -473,7 +475,7 @@ func (cs *supply) allocateMemory(cr *request) (memoryMap, error) {
 		available := cs.mem[memoryPMEM] - cs.grantedMem[memoryPMEM]
 
 		log.Debug("%s: trying %s from PMEM, available %s",
-			cr.GetContainer().PrettyName(),
+			r.GetContainer().PrettyName(),
```

Comment:
This is a matter of taste, but I'd think this code would be less fragile if we'd have the same reservation code routine for every memory type:

```
for mem in [memoryPMEM, memoryDRAM, memoryHBM]:
    if remaining == 0: break
    if remaining < available:
        ...
    else:
        ...
```

Sure, it'll need the coldstart check after PMEM reservation inside the loop, but I'd still think it would make this easier to read and test.

WDYT?

Reply:
Yes, I agree. Especially, since related code we've fixed or added in the scope of this and other #609-related PRs were updated and written like that. I'll update this one, too.

Reply:
Refactored `allocateMemory()`. Still passes unit tests.